go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -799,8 +799,8 @@ func isLayer2UserDefinedPrimaryNetwork(netInfo util.NetInfo) bool {
 	return netInfo.IsPrimaryNetwork() && netInfo.TopologyType() == types.Layer2Topology
 }
 
-func autoExcludeCIDRs(subnet *net.IPNet) []*net.IPNet {
-	gwIP := util.GetNodeGatewayIfAddr(subnet).IP
+func autoExcludeCIDRs(subnet *net.IPNet, netInfo util.NetInfo) []*net.IPNet {
```

Comment:
if we're now passing the `NetInfo` struct, I think it's redundant to pass the subnet. 

Shouldn't we drop that now ? We can access the subnet from the `netInfo`.